src/services/sync/strategies/consumerActions/UserAction.js

```diff
@@ -0,0 +1,63 @@
+const _ = require('lodash');
+
+const BaseConsumerAction = require('./BaseConsumerAction');
+// TODO: place from where it is importat must be fixed later
+const { LDAP_SYNC_ACTIONS } = require('../LDAPSyncer');
+const { SchoolRepo, UserRepo } = require('../../repo');
+
+const defaultOptions = {
+	allowedLogKeys: ['ldapId', 'systemId', 'roles', 'activated'],
+};
+
+class UserAction extends BaseConsumerAction {
+	constructor(filterActive = true, options = defaultOptions) {
+		super(LDAP_SYNC_ACTIONS.SYNC_USER, options);
+		this.filterActive = filterActive;
+	}
+
+	async action(data = {}) {
+		const { user = {}, account = {} } = data;
+		const school = await SchoolRepo.findSchoolByLdapIdAndSystem(user.schoolDn, user.systemId);
+		if (school) {
+			const foundUser = await UserRepo.findByLdapIdAndSchool(user.ldapId, school._id);
+			if (foundUser !== null) {
+				await this.updateUserAndAccount(foundUser, user, account);
+			} else {
+				await this.createUserAndAccount(user, account, school._id);
+			}
+		}
+	}
+
+	async updateUserAndAccount(foundUser, user, account) {
+		const updateObject = {};
+		if (user.firstName !== foundUser.firstName) {
+			updateObject.firstName = user.firstName || ' ';
+		}
+		if (user.lastName !== foundUser.lastName) {
+			updateObject.lastName = user.lastName;
+		}
+		// Updating SchoolId will cause an issue. We need to discuss about it
```

Comment:
shouldn't this fail when the schoolId changes? why ignore it here in the update?

Reply:
this comment was taken from the initial implementation. But we will look on that case more intensively.

Reply:
Since we search for the user by schoolId, we will not find the user if the schoolId has changed.

Reply:
the schoolId update is not possible. It will cause User already exists error.

Reply:
do we need to ensure school id has not changed? when it has changed, isn't it good that this would fail?